pkg/controller/jobframework/reconciler.go

```diff
@@ -300,7 +306,14 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 
 // startJob will unsuspend the job, and also inject the node affinity.
 func (r *JobReconciler) startJob(ctx context.Context, job GenericJob, object client.Object, wl *kueue.Workload) error {
-	nodeSelectors, err := r.getNodeSelectors(ctx, wl)
+	log := ctrl.LoggerFrom(ctx)
+	//get the original selectors and store them in the job object
+	originalSelectors := r.getNodeSelectorsFromPodSets(wl)
+	if err := nodeSelectorsSetToObject(object, originalSelectors); err != nil {
+		log.V(3).Error(err, "Unable to set original node selectors")
```

Comment:
Shouldn't we return with error here? I'm concerned that we may start the job and later not be able to restore the node selectors.

Reply:
done